### PR TITLE
chore(horologium): clean up code style and clearer trigger job logic

### PR DESCRIPTION
**Pull request by sxllwx**

### CODE REVIEW

**Comment by sxllwx** [comment]
Can anyone help me?

**Review by matthyx — APPROVED** [review]
/lgtm

File: prow/cmd/horologium/main.go [hunk 1]
```diff
@@ -101,24 +101,24 @@ func main() {
...
-		logrus.WithError(err).Fatal("failed to get a prowjob informer")
+		logrus.WithError(err).Fatal("Failed to get a prowjob informer")
 	}
 	interrupts.Run(func(ctx context.Context) {
 		if err := cluster.Start(ctx); err != nil {
-			logrus.WithError(err).Fatal("Cache failed to start")
+			logrus.WithError(err).Fatal("Cluster failed to start")
 		}
-		logrus.Info("Cache finished gracefully.")
+		logrus.Info("Cluster finished gracefully.")
```
Comment (cjwagner) [inline prow/cmd/horologium/main.go:116]: I think the original text was more correct. This now reads as if we were spinning up a new cluster, but we're actually creating a controller-runtime manager (informer?) to monitor the cluster.

File: prow/cmd/horologium/main.go [hunk 2]
```diff
@@ -180,39 +181,34 @@ func sync(prowJobClient ctrlruntimeclient.Client, cfg *config.Config, cr cronCli
...
 					"previous-found": previousFound,
 					"should-trigger": shouldTrigger,
 					"name":           p.Name,
 					"job":            p.JobBase.Name,
-				}).Info("skipping cron periodic")
+				}).Info("Skipping cron periodic")
+			}
+			continue
+		}
+		if shouldTrigger {
```
Comment (cjwagner) [inline prow/cmd/horologium/main.go:197]: We don't need this conditional. If `should-trigger` is false that seems like something important to know while debugging.

File: prow/cmd/horologium/main.go [hunk 3]
```diff
@@ -180,39 +181,34 @@ func sync(prowJobClient ctrlruntimeclient.Client, cfg *config.Config, cr cronCli
...
+			logger = logger.WithField("should-trigger", shouldTrigger)
+		}
+		if !previousFound || shouldTrigger {
+			prowJob := pjutil.NewProwJob(pjutil.PeriodicSpec(p), p.Labels, p.Annotations)
+			prowJob.Namespace = cfg.ProwJobNamespace
+			logger.WithFields(pjutil.ProwJobFields(&prowJob)).Info("Triggering new run of interval periodic.")
```
Comment (cjwagner) [inline prow/cmd/horologium/main.go:203]: We no longer know if this is an interval of cron periodic so we should omit that.

**Review by cjwagner — COMMENTED** [review]
I just have some nits, the refactor looks right.

File: prow/cmd/horologium/main.go (hunk 1 above)
Comment (sxllwx) [reply prow/cmd/horologium/main.go]: Okay, but we can consider using controller/manager

File: prow/cmd/horologium/main.go (hunk 3 above)
Comment (sxllwx) [reply prow/cmd/horologium/main.go]: Added related fixes. PTAL

**Comment by sxllwx** [comment]
Thanks for your time @cjwagner , I've addressed and fixed Comments, PTAL

**Review by cjwagner — APPROVED** [review]
LGTM, but I'd prefer to avoid these kind of changes in the future.  Horologium hasn't really been touched in years and has been working well. This provides no functional improvement and could introduce a regression if we make a mistake. Additionally this is not an area where technical debt is causing toil so these kind of changes are probably not the best use of our time.

**Pull request merged**